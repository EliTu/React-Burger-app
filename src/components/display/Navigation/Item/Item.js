import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { signInToggleClick } from '../AuthItems/store/actions';
import { authSignout } from '../../../containers/Authentication/store/actions';
import Icon from '../../../UI/Icon/Icon';
import { faSignInAlt } from '@fortawesome/free-solid-svg-icons';
import styles from './Item.module.css';
import PropTypes from 'prop-types';

const Item = ({
	link,
	children,
	isSignInDisplayed,
	signInItem,
	toggleSignIn,
	signOutItem,
	signOutClicked,
	isLoggedIn,
}) => {
	// CSS Modules styles:
	const { Item, active, AuthSignin, SigninActive } = styles;

	// Handle toggling the display property of the LogIn component
	const handleSignInToggleClick = event => {
		event.preventDefault();
		if (!isSignInDisplayed) toggleSignIn();
	};

	// Handle loging user out upon click on 'Log out':
	const handleLogOutClick = event => {
		event.preventDefault();
		signOutClicked('signOut');
	};

	// Set Sign in tab custom styles:
	const signInDefaultStyle = !isLoggedIn ? AuthSignin : null;
	const signInActiveStyle = isSignInDisplayed ? SigninActive : null;

	const authType =
		signInItem || signOutItem ? (
			<a
				className={[Item, signInDefaultStyle, signInActiveStyle].join(
					' '
				)}
				href={link}
				onClick={event =>
					signInItem
						? handleSignInToggleClick(event)
						: handleLogOutClick(event)
				}
			>
				{signInItem && <Icon iconType={faSignInAlt} />}
				{signInItem ? 'Sign in' : 'Log out'}
			</a>
		) : (
			<NavLink className={Item} activeClassName={active} to={link} exact>
				{children}
			</NavLink>
		);

	return <div className={Item}>{authType}</div>;
};

Item.propTypes = {
	link: PropTypes.string.isRequired,
	children: PropTypes.node.isRequired,
	signInItem: PropTypes.bool,
	isSignInDisplayed: PropTypes.bool,
	toggleSignIn: PropTypes.func,
	signOutClicked: PropTypes.func,
	isLoggedIn: PropTypes.bool,
};

// Redux setup:
const mapStateToProps = state => {
	return {
		isSignInDisplayed: state.signIn.isSignInDisplayed,
		isLoggedIn: state.auth.isLoggedIn,
	};
};
const mapDispatchToProps = dispatch => {
	return {
		toggleSignIn: () => dispatch(signInToggleClick()),
		signOutClicked: signOut => dispatch(authSignout(signOut)),
	};
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Item);
