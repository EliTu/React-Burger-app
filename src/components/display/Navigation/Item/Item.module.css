@import '../../../../variables.css';

.Item {
	box-sizing: border-box;
	font-size: 1.3rem;
	margin: 0.8rem auto;
	width: 75%;
	display: block;
	transition: 0.5s all;
}

.Item a {
	width: 100%;
	align-self: auto;
	color: var(--coloring-light);
	text-decoration: none;
	text-align: center;
	display: block;
}

.Item .AuthSignin {
	color: var(--glow-green-darker);
}

.Item a:hover {
	border-bottom: 5px solid var(--coloring-bright-light);
	background-color: rgba(219, 221, 219, 0.144);
}

.Item a:active,
.Item a.active {
	background-color: rgba(0, 0, 0, 0);
	border-bottom: 10px solid var(--coloring-bright-light);
	color: var(--coloring-bright-light);
	background-color: rgba(219, 221, 219, 0.144);
}

/* Large screen */
@media screen and (min-width: 500px) {
	.Item {
		font-size: initial;
		margin: 0;
		height: 100%;
		width: auto;
		display: flex;
		align-items: center;
		transition: 0.3s all;
	}

	.Item a {
		align-self: flex-end;
		font-size: 1rem;
		align-self: auto;
		color: var(--coloring-light);
		height: 100%;
		width: auto;
		text-align: center;
		padding: 0.9rem 1rem;
		border-bottom: 4px solid transparent;
		border-radius: 0;
	}

	.Item .SigninActive {
		color: rgb(99, 248, 99);
		border-bottom: 10px solid var(--coloring-bright-light);
	}
}

/*  Medium screne */
@media only screen and (max-width: 850px) {
	.Item a {
		font-size: 0.9rem;
		width: 100%;
		padding: 0.9rem 0.4rem;
	}
}

/* small screen */
@media screen and (max-width: 550px) {
	.Item a {
		font-size: 1rem;
		width: 100%;
		padding: 0.9rem 0.4rem;
	}

	.Item a:active,
	.Item a.active {
		background-color: rgba(0, 0, 0, 0);
		border-bottom: 3px solid var(--glow-green-darker);
		color: var(--coloring-bright-light);
	}
}