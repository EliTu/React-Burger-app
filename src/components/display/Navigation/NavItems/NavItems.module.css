.NavItems {
    margin: 0 1rem;
    padding: 0;
    list-style: none;
    display: flex;
    align-content: space-between;
    flex-direction: column;
    align-items: center;
    height: 100%;
}

/* large screen */
@media only screen and (min-width: 500px) {
    .NavItems {
        flex-flow: row;
        width: 100%;
    }
}