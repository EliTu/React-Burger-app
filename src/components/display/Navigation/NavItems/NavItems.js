import React from 'react';
import { connect } from 'react-redux';
import Item from '../Item/Item';
import Icon from '../../../UI/Icon/Icon';
import styles from './NavItems.module.css';
import {
	faStream,
	faList,
	faShoppingCart,
	faBookmark,
} from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';

export const NavItems = ({ isLoggedIn }) => {
	// CSS Module styles:
	const { NavItems } = styles;

	return (
		<div className={NavItems}>
			<Item link="/">
				{' '}
				<Icon iconType={faStream} />
				Burger Builder
			</Item>
			{isLoggedIn && (
				<Item link="/checkout">
					<Icon iconType={faShoppingCart} />
					Checkout
				</Item>
			)}
			<Item link="/orders">
				{' '}
				<Icon iconType={faList}> </Icon>Orders
			</Item>
			<Item link="/about">
				{' '}
				<Icon iconType={faBookmark} />
				About
			</Item>
		</div>
	);
};

NavItems.propTypes = {
	isLoggedIn: PropTypes.bool,
};

// Redux Setup:
const mapStateToProps = state => {
	return {
		isLoggedIn: state.auth.isLoggedIn,
	};
};

export default connect(mapStateToProps)(NavItems);
