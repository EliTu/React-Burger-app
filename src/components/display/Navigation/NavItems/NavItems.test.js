import React from 'react';
import { shallow } from 'enzyme';
import { NavItems } from './NavItems';
import Item from '../Item/Item';

describe('<NavItems/>', () => {
	let component;
	beforeEach(() => {
		component = shallow(<NavItems />);
	});

	it('should render 3 nav <Item/> components (as long as user not authenticated)', () => {
		expect(component).toMatchSnapshot();
		expect(component.find(Item)).toHaveLength(3);
		expect(component.find(Item)).not.toHaveLength(2);
		expect(component.find(Item)).not.toBeFalsy();
	});

	it('should render 4 nav <Item/> components if the user is authenticated and logged in', () => {
		component.setProps({ isLoggedIn: true });

		expect(component).toMatchSnapshot();
		expect(component.find(Item)).toHaveLength(4);
		expect(component.find(Item)).not.toHaveLength(3);
		expect(component.find(Item)).not.toBeFalsy();
	});
});
