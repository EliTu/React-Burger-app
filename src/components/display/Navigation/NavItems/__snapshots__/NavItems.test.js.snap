// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItems/> should render 3 nav <Item/> components (as long as user not authenticated) 1`] = `
<div
  className="NavItems"
>
  <Connect(Item)
    link="/"
  >
     
    <Icon
      iconType={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f550",
            "M16 128h416c8.84 0 16-7.16 16-16V48c0-8.84-7.16-16-16-16H16C7.16 32 0 39.16 0 48v64c0 8.84 7.16 16 16 16zm480 80H80c-8.84 0-16 7.16-16 16v64c0 8.84 7.16 16 16 16h416c8.84 0 16-7.16 16-16v-64c0-8.84-7.16-16-16-16zm-64 176H16c-8.84 0-16 7.16-16 16v64c0 8.84 7.16 16 16 16h416c8.84 0 16-7.16 16-16v-64c0-8.84-7.16-16-16-16z",
          ],
          "iconName": "stream",
          "prefix": "fas",
        }
      }
    />
    Burger Builder
  </Connect(Item)>
  <Connect(Item)
    link="/orders"
  >
     
    <Icon
      iconType={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f03a",
            "M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z",
          ],
          "iconName": "list",
          "prefix": "fas",
        }
      }
    >
       
    </Icon>
    Orders
  </Connect(Item)>
  <Connect(Item)
    link="/about"
  >
     
    <Icon
      iconType={
        Object {
          "icon": Array [
            384,
            512,
            Array [],
            "f02e",
            "M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z",
          ],
          "iconName": "bookmark",
          "prefix": "fas",
        }
      }
    />
    About
  </Connect(Item)>
</div>
`;

exports[`<NavItems/> should render 4 nav <Item/> components if the user is authenticated and logged in 1`] = `
<div
  className="NavItems"
>
  <Connect(Item)
    link="/"
  >
     
    <Icon
      iconType={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f550",
            "M16 128h416c8.84 0 16-7.16 16-16V48c0-8.84-7.16-16-16-16H16C7.16 32 0 39.16 0 48v64c0 8.84 7.16 16 16 16zm480 80H80c-8.84 0-16 7.16-16 16v64c0 8.84 7.16 16 16 16h416c8.84 0 16-7.16 16-16v-64c0-8.84-7.16-16-16-16zm-64 176H16c-8.84 0-16 7.16-16 16v64c0 8.84 7.16 16 16 16h416c8.84 0 16-7.16 16-16v-64c0-8.84-7.16-16-16-16z",
          ],
          "iconName": "stream",
          "prefix": "fas",
        }
      }
    />
    Burger Builder
  </Connect(Item)>
  <Connect(Item)
    link="/checkout"
  >
    <Icon
      iconType={
        Object {
          "icon": Array [
            576,
            512,
            Array [],
            "f07a",
            "M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z",
          ],
          "iconName": "shopping-cart",
          "prefix": "fas",
        }
      }
    />
    Checkout
  </Connect(Item)>
  <Connect(Item)
    link="/orders"
  >
     
    <Icon
      iconType={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f03a",
            "M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z",
          ],
          "iconName": "list",
          "prefix": "fas",
        }
      }
    >
       
    </Icon>
    Orders
  </Connect(Item)>
  <Connect(Item)
    link="/about"
  >
     
    <Icon
      iconType={
        Object {
          "icon": Array [
            384,
            512,
            Array [],
            "f02e",
            "M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z",
          ],
          "iconName": "bookmark",
          "prefix": "fas",
        }
      }
    />
    About
  </Connect(Item)>
</div>
`;
